python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -1,98 +1,138 @@
-# 
+#
 # LSST Data Management System
 #
 # Copyright 2008-2017  AURA/LSST.
-# 
+#
 # This product includes software developed by the
 # LSST Project (http://www.lsst.org/).
 #
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation, either version 3 of the License, or
 # (at your option) any later version.
-# 
+#
 # This program is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
-# 
-# You should have received a copy of the LSST License Statement and 
-# the GNU General Public License along with this program.  If not, 
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
 # see <https://www.lsstcorp.org/LegalNotices/>.
 #
 from __future__ import absolute_import, division, print_function
 
-__all__ = ["BaseSourceSelectorConfig", "BaseSourceSelectorTask", "sourceSelectorRegistry"]
-
 import abc
 
 import lsst.afw.table as afwTable
 import lsst.pex.config as pexConfig
 import lsst.pipe.base as pipeBase
 from future.utils import with_metaclass
 
+__all__ = ["BaseSourceSelectorConfig", "BaseSourceSelectorTask",
+           "sourceSelectorRegistry"]
+
 
 class BaseSourceSelectorConfig(pexConfig.Config):
-    badFlags = pexConfig.ListField(
```

Comment:
Sorry it wasn't clearer. This is a implementation ticket for RFC-198 so we are changing the API. This ticket is mainly to create and test the new base class. The two selectors you mention will be updated in DM-9835 and DM-10797 respectively.